crawl-ref/source/dat/des/variable/float.des

```diff
@@ -4901,6 +4901,77 @@ MAP
 .cccccccccc
 ENDMAP
 
+NAME:   dreamdust_black_sun_cult
+TAGS:   transparent no_monster_gen
+ORIENT: float
+DEPTH:  D:6-10, Zot, !Zot:$
+WEIGHT: 3
+COLOUR: bm = red
+TILE:   m = dngn_transparent_wall_red
+FTILE:  -mA? = floor_volcanic
+KFEAT:  ! = .
+MARKER: ! = lua:fog_machine { cloud_type='flame', pow_min = 10, pow_max = 10, \
+                              delay = 10, size = 1, excl_rad = 0 }
+: if you.in_branch("D") then
+KMONS:  ? = sun demon
+KMONS:  - = nothing w:50 / demonspawn
+KITEM:  ? = manual of fire magic w:5 / \
+            randbook disc:fire disc2:necromancy spells:scorch \
+            slevels:8 numspells:4 title:The_Black_Sun, \
+            ring of protection from fire / ring of fire / \
+            staff of fire / scroll of immolation q:3
+: else
+KMONS:  ? = orb of fire
+KMONS:  - = nothing w:50 / black sun / \
+            draconian monk w:5 / draconian scorcher
+KITEM:  ? = manual of fire magic w:5 / \
+            randbook disc:fire disc2:necromancy spells:irradiate \
+            slevels:25 numspells:4 title:The_Black_Sun, \
+            ring of protection from fire randart / ring of fire randart
+: end
+MAP
+ .........
+..!bbbbb!..
+.!bb---bb!.
+.bb-----bb.
+.b--mAm--b.
+.b--m?m--b.
+.b--mmm--b.
+.bb-----bb.
+.!bb---bb!.
+..!bb+bb!..
+ .........
+ENDMAP
+
+NAME:    dreamdust_cat_pyramid
+TAGS:    no_monster_gen transparent no_rotate
+DEPTH:   Crypt, !Crypt:$, Depths, !Depths:$
+ORIENT:  float
+WEIGHT:  3
+TILE:    c = wall_tomb
+COLOUR:  c = yellow
+FTILE:   -S1+ = floor_sandstone
+KFEAT:   S = granite_statue
+TILE:    S = dngn_sarcophagus_pedestal_left
+TILE:    G = dngn_statue_cat
+KITEM:   $ = gold q:100
```

Comment:
It's better to have randomization by using normal gold piles. If you need more gold than three piles would typically general, simply put more gold in additional places. In general gold only loot is less interesting as vault loot for a player this late, so I'd actually recommend introducing some good general or consumable loot in the form of `%` (normal quality random), `*` (higher quality random), or `|` (higher quality mostly equipment). Placing these in front of your cat statue will theme as being "offerings", since that seems to be what you're establishing. Alternately you can use the tables `dgn.loot_scrolls` and `dgn.loot_potions` to make higher quality consumable potion and scroll loot, if you like that more thematically. Gold only is fine if you strongly prefer this, but simply use more `$` instead of defining custom gold counts here, if you go that route.